book/src/saving_window_state.md

```diff
@@ -20,6 +20,10 @@ First, we create one and add methods for getting and setting the window state.
 
 ```rust,no_run
 {{#rustdoc_include ../listings/saving_window_state/1/custom_window/mod.rs:mod}}
+
+# // Please ignore this line
+# // It is only there to make rustdoc happy
```

Comment:
why rustdoc is even run on the book ?

Reply:
I think it just mimics the behavior of rustdoc. It wraps every code block without main in a new main.